include/deal.II/base/thread_management.h

```diff
@@ -1833,10 +1833,8 @@ namespace Threads
     void
     join_all() const
     {
-      for (typename std::list<Task<RT>>::const_iterator t = tasks.begin();
-           t != tasks.end();
-           ++t)
-        t->join();
+      for (auto t : tasks)
```

Comment:
Shouldn't this be a reference type?

Reply:
Yes, although it happens to work because all task objects corresponding to the same task point to a single instance of a task base object :-)